client/components/drop-zone/index.jsx

```diff
@@ -85,29 +75,29 @@ module.exports = React.createClass( {
 				subtree: true
 			} );
 		}
-	},
+	}
 
-	disconnectMutationObserver: function() {
+	disconnectMutationObserver() {
 		if ( ! this.observer ) {
 			return;
 		}
 
 		this.observer.disconnect();
 		delete this.observer;
-	},
+	}
 
-	detectNodeRemoval: function( mutations ) {
+	detectNodeRemoval( mutations ) {
 		mutations.forEach( ( mutation ) => {
 			if ( ! mutation.removedNodes.length ) {
 				return;
 			}
 
 			this.dragEnterNodes = without( this.dragEnterNodes, Array.from( mutation.removedNodes ) );
 		} );
-	},
+	}
 
-	toggleDraggingOverDocument: function( event ) {
-		var isDraggingOverDocument, detail, isValidDrag;
+	toggleDraggingOverDocument( event ) {
+		let isDraggingOverDocument, detail, isValidDrag;
```

Comment:
Not necessarily the responsibility of an initial refactor, but we could start moving the declarations away from top-of-file, e.g. `const isDraggingOverElement = isVali/* ... */;` further down in the function.